pupil_src/shared_modules/surface_tracker/surface_online.py

```diff
@@ -0,0 +1,51 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2018 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import collections
+
+from surface_tracker.surface import Surface
+
+
+class Surface_Online(Surface):
+    """Surface_Online recalculates it's location on demand."""
+
+    def __init__(self, name="unknown", init_dict=None):
+        super().__init__(name=name, init_dict=init_dict)
+
+        self.gaze_history_length = 1
```

Comment:
If you don't intend to change this, it would be better to make it a class variable `GAZE_HISTORY_LENGTH` (aka constant).